lib/backend/k8s/resources/watcher.go

```diff
@@ -128,14 +121,10 @@ func (crw *k8sWatcherConverter) processK8sEvents() {
 				case crw.resultChan <- *e:
 					crw.logCxt.Debug("Kubernetes event converted and sent to backend watcher")
 
-					// If this is an error event, check to see if it's a terminating one (the
-					// convertEvent method will decide that).  If so, terminate this watcher.
+					// If this is an error event, terminate the watcher and bubble up the watcher error
 					if e.Type == api.WatchError {
-						crw.logCxt.WithError(e.Error).Debug("Watch event was an error event type")
-						if _, ok := e.Error.(cerrors.ErrorWatchTerminated); ok {
-							crw.logCxt.Debug("Watch event indicates a terminated watcher")
-							return
-						}
+						crw.logCxt.Debug("Watch event was an error event type")
```

Comment:
Same here - do we need to terminate.  I think we can just bubble up errors without terminating the watcher.  If the channel gets closed then we'll terminate the watcher through that mechanism.

Reply:
Sure, I will update to bubble up the error.